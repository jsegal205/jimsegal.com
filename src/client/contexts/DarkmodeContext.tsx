import React, {
  useContext,
  useState,
  FunctionComponent,
  ReactNode,
} from "react";

type DarkmodeProviderProps = {
  children: ReactNode;
};
type DarkmodeContextTypes = { darkmode: boolean; toggleDarkmode: () => void };

const DarkmodeContext = React.createContext({
  darkmode: false,
  toggleDarkmode: () => {},
});

const useDarkModeContext: () => DarkmodeContextTypes = () =>
  useContext(DarkmodeContext);

const DarkmodeProvider: FunctionComponent<DarkmodeProviderProps> = ({
  children,
}: DarkmodeProviderProps) => {
  const [darkmode, setDarkmode] = useState(false);
  const toggleDarkmode = () => setDarkmode(!darkmode);

  return (
    <DarkmodeContext.Provider value={{ darkmode, toggleDarkmode }}>
      {children}
    </DarkmodeContext.Provider>
  );
};

export { DarkmodeProvider, useDarkModeContext };
