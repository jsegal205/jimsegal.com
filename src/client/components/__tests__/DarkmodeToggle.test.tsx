import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import DarkModeToggle from "../DarkmodeToggle";
import { DarkmodeProvider } from "../../contexts/DarkmodeContext";

describe("<DarkmodeToggle />", () => {
  beforeEach(() => {
    render(
      <DarkmodeProvider>
        <DarkModeToggle />
      </DarkmodeProvider>
    );
  });

  describe("when component is initialized", () => {
    it("then sets the light class by default", () => {
      expect(screen.getByText(/Dark/i)).toBeTruthy();
      expect(document.body.className).toEqual("");
    });
  });

  describe("when the toggle theme input is clicked", () => {
    beforeEach(() => {
      userEvent.click(screen.getByText(/Dark/i));
    });

    it("then sets the dark class", () => {
      expect(screen.getByText(/Light/i)).toBeTruthy();
      expect(document.body.className).toEqual("dark");
    });
  });
});
