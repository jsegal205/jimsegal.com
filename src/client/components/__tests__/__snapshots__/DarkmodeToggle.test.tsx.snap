// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DarkmodeToggle /> renders correct structure 1`] = `
<div>
  <div
    class="toggle"
  >
    <label
      class="switch"
    >
      <input
        type="checkbox"
      />
      <div
        class="slider round"
      >
        DARK?
      </div>
    </label>
  </div>
</div>
`;
